### PR TITLE
Fix package installation for ros2

### PR DESCRIPTION
**Pull request by aosmw**
This is not yet ready but might be worth more eyes.

I install the ortools into opt/f2c_ortools
I tweak the fields2coverConfig.cmake to find it.

You may not want the change I did to pull spline.h into the repository rather than use it through FetchContent.  
It does some packaging stuff that I don't think is correct.
It is a header only library.

The python/swig setup is needs help to find the ortools libraries.

```
export LD_LIBRARY_PATH=/opt/ros/rolling/opt/f2c_ortools/lib
$python3
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import fields2cover
```

### CODE REVIEW

**Comment by aosmw** [comment]
Requires https://github.com/Fields2Cover/ortools_vendor/pull/3 to be merged and bloom released.

**Comment by aosmw** [comment]
https://github.com/Fields2Cover/ortools_vendor/pull/3 got merged but https://build.ros2.org has been down since 8th June 2024 1:37pm so there is no package of ortools_vendor to allow the next step.

image

**Comment by Gonzalo-Mier** [comment]
It back on. You can see its progress on https://build.ros2.org/job/Rbin_uN64__ortools_vendor__ubuntu_noble_amd64__binary/44/console

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,9 +25,9 @@ option(BUILD_TESTS "Build tests" ON)
 option(BUILD_PYTHON "Build Python SWIG module" ON)
 option(BUILD_DOC "Build Documentation" OFF)
 option(BUILD_SHARED_LIBS "Build shared library(.so)" ON)
-option(USE_ORTOOLS_RELEASE "Get or-tools from release tarball" ON)
+option(USE_ORTOOLS_RELEASE "Get or-tools from release tarball" OFF)
 option(USE_ORTOOLS_FETCH_SRC "Get or-tools from source" OFF)
-option(USE_ORTOOLS_VENDOR "Get or-tools from ortools_vendor" OFF)
+option(USE_ORTOOLS_VENDOR "Get or-tools from ortools_vendor" ON)
```
Comment (aosmw) [inline CMakeLists.txt:30]: To behave as a ROS package built from source that is easily used by colcon builds in a wider ros environment we need to default USE_ORTOOLS_VENDOR to ON.

If you don't want that, then the option needs to at least be set on for ROS2 packaging builds. i.e. Like it was done for FETCHCONTENT_FULLY_DISCONNECTED=OFF
https://github.com/ros2-gbp/fields2cover-release/blob/patches/debian/rolling/fields2cover/0002-Customized-debian-rules-file.patch  

I don't know how you actually did that.  Did you have to visit all the active branches?

File: cmake/Fields2CoverConfig.cmake.in [hunk 2]
```diff
@@ -4,4 +4,9 @@ include("${CMAKE_CURRENT_LIST_DIR}/Fields2Cover-Targets.cmake" )
 
 include(CMakeFindDependencyMacro)
 find_dependency(GDAL 3.0 REQUIRED)
-find_dependency(ortools REQUIRED)
+
+list(INSERT CMAKE_PREFIX_PATH 0 "${PACKAGE_PREFIX_DIR}/opt/ortools_vendor")
+list(INSERT CMAKE_PREFIX_PATH 1 "${PACKAGE_PREFIX_DIR}/opt/f2c_ortools")
+
+find_dependency(ortools REQUIRED NO_SYSTEM_ENVIRONMENT_PATH)
```
Comment (aosmw) [inline cmake/Fields2CoverConfig.cmake.in:11]: Am not sure if we should have NO_SYSTEM_ENVIRONMENT_PATH here.

File: cmake/Fields2CoverConfig.cmake.in [hunk 3]
```diff
@@ -4,4 +4,9 @@ include("${CMAKE_CURRENT_LIST_DIR}/Fields2Cover-Targets.cmake" )
 
 include(CMakeFindDependencyMacro)
 find_dependency(GDAL 3.0 REQUIRED)
-find_dependency(ortools REQUIRED)
+
+list(INSERT CMAKE_PREFIX_PATH 0 "${PACKAGE_PREFIX_DIR}/opt/ortools_vendor")
+list(INSERT CMAKE_PREFIX_PATH 1 "${PACKAGE_PREFIX_DIR}/opt/f2c_ortools")
```
Comment (aosmw) [inline cmake/Fields2CoverConfig.cmake.in:9]: This *should* allow the ortools to be found in the cases where they are provided by USE_ORTOOLS_VENDOR or USE_ORTOOLS_RELEASE.  Preference for ortools_vendor version.

**Review by aosmw — COMMENTED** [review]
@Gonzalo-Mier - Ready for review.

**Comment by Gonzalo-Mier** [comment]
Thank you really much for the PR. I'm not completely sure how easy is for non-ROS-users to install F2C now, mainly due to ortools_vendors.

**Pull request merged**